### PR TITLE
Add xslt

### PR DESCRIPTION
**Pull request by TanguyGen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by TanguyGen** [comment]
test

File: recipes/r-xslt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    - {{ posix }}sed               # [win]
+    - {{ posix }}grep              # [win]
+    - {{ posix }}autoconf
+    - {{ posix }}automake          # [not win]
+    - {{ posix }}automake-wrapper  # [win]
+    - {{ posix }}pkg-config
+    - {{ posix }}make
+    - {{ posix }}coreutils         # [win]
+    - {{ posix }}zip               # [win]
+    - libxslt
```
Comment (mfansler) [inline recipes/r-xslt/meta.yaml:38]: I don't think this should be needed.

```suggestion
```

File: recipes/r-xslt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
+  host:
+    - r-base
+    - r-rcpp
+    - libxslt
+    - r-xml2 >=1.3.0
```
Comment (mfansler) [inline recipes/r-xslt/meta.yaml:44]: This package itself links against `libxml2`, so we need to declare that. Also, let's reorder:

```suggestion
    - r-xml2 >=1.3.0
    - libxml2
    - libxslt
```

File: recipes/r-xslt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+test:
+  commands:
+    - $R -e "library('xslt')"           # [not win]
+    - "\"%R%\" -e \"library('xslt')\""  # [win]
+
+about:
+  home: https://ropensci.r-universe.dev/xslt https://docs.ropensci.org/xslt/
+  license: GPL-2.0-only
```
Comment (mfansler) [inline recipes/r-xslt/meta.yaml:59]: License show `GPL (>= 3)` so let's stick with that. Also, split the URLs semantically:

```suggestion
  home: https://ropensci.r-universe.dev/xslt
  doc_url: https://docs.ropensci.org/xslt/
  license: GPL-2.0-or-later
```

File: recipes/r-xslt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  summary: An extension for the 'xml2' package to transform XML documents by applying an 'xslt'
+    style-sheet.
+  license_family: GPL2
+  license_file:
+    - '{{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2'
```
Comment (mfansler) [inline recipes/r-xslt/meta.yaml:64]: Include expanded license:

```suggestion
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-3
```

File: recipes/r-xslt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+source:
+  url:
+    - {{ cran_mirror }}/src/contrib/xslt_{{ version }}.tar.gz
+    - {{ cran_mirror }}/src/contrib/Archive/xslt/xslt_{{ version }}.tar.gz
+  sha256: 64aacaff2dacb474c3abe2e91589676f8741981011ca7fc35b814e0ceb884c31
+
+build:
+  merge_build_host: True  # [win]
+  number: 0
```
Comment (mfansler) [inline recipes/r-xslt/meta.yaml:17]: Let's skip Windows. The package source will need to be patched to get it working and this is something that can be postponed until after feedstock creation.

```suggestion
  skip: True  # [win]
  number: 0
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for submitting this! Some adjustments are need to get it working.

**Comment by TanguyGen** [comment]
You really are a life savior ;) Thank you !

**Comment by mfansler** [comment]
I'm not sure if this is related to the **osx** failure, but I did notice it links against `liblzma`, so I added `xz` as a dependency.

**Comment by bgruening** [comment]
Ok, now we are at:

> Error: package or namespace load failed for ‘xslt’ in dyn.load(file, DLLpath = DLLpath, ...):
 Warning: program compiled against libxml 211 using older 210

File: recipes/r-xslt/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    - r-xml2 >=1.3.0
+    - libxml2
+    - libxslt
+    - xz                           # [osx]
+  run:
+    - r-base
+    - {{ native }}gcc-libs         # [win]
+    - r-rcpp
+    - r-xml2 >=1.3.0
+    - libxslt
```
Comment (bgruening) [inline recipes/r-xslt/meta.yaml:52]:
```suggestion
```

It has run_exports defined: https://github.com/conda-forge/libxslt-feedstock/blob/main/recipe/meta.yaml#L16

Comment (mfansler) [reply recipes/r-xslt/meta.yaml:52]: Good to know!

**Comment by mfansler** [comment]
> Ok, now we are at:
> 
> > Error: package or namespace load failed for ‘xslt’ in dyn.load(file, DLLpath = DLLpath, ...):
> > Warning: program compiled against libxml 211 using older 210

Possibly linking against the build environment's copy(?), since the other environments should be taking the pinned version (v2.10). There's probably a more principled workaround (e.g., removing pkg-config entries in build), but for now let's try forcing the v2.10 in build env. We could drop this once the `libxml2110` migration hits this in feedstock.

**Review by mfansler — APPROVED** [review]

**Comment by mfansler** [comment]
@bgruening I'm satisfied with this, but I'll leave you with final call. That is, let me know if you'd prefer we modify the `build.sh` to purge pkg-config entries from build environment instead of the workaround I did.

**Comment by bgruening** [comment]
I'm ok with it, thanks for your support.

**Pull request merged**